### ISSUE TITLE
[Bug] force_batch deletes data from model_tmp_not_partitioned before coping to the final table

### ISSUE THREAD

**Issue by juanjodevio** [dbt-athena/dbt-athena#618]
### Is this a new bug in dbt-athena?

- [X] I believe this is a new bug in dbt-athena
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

Using dbt athena 1.7.2 with a model set with the configs listed below as first step creates a table `model_tmp_not_partitioned` and data appears in s3 under the `external_location` we have configured but moments later s3 objects disappear and following select statements for `force_batch` fails due to data missing from the s3 path.

Then I tried using `force_batch: false` and worked fine
```
models:
  my_project:
    +force_batch: true
    +table_type: "iceberg"
    +format: "parquet"
    +write_compression: "snappy"
    +on_schema_change: "sync_all_columns"
```

### Expected Behavior

`force_batch` successfully creates table with partition

### Environment

```markdown
- OS: MacOS 14.4.1
- Python: 3.11
- dbt: 1.7.10
- dbt-athena-community: 1.7.2
```

**Comment by nicor88** [comment]
The issue is due to the fact that you are using an external location with iceberg materialization table. This was not allowed since 1.7.1,and due to a regression we introduced in 1.7.2.

Long story short: no external location allowed for iceberg at the moment. And only unique s3 data naming.
Try to use s3_data_naming=table_unique and see if all works fine.

**Comment by nicor88** [comment]
Closing because the issue is being fixed in dbt-athena-community==1.8.0b1, and it was due to https://github.com/dbt-athena/dbt-athena/issues/597
